api/controllers/relationships.js

```diff
@@ -20,6 +24,51 @@ function getAllRelationships(req, res) {
     });
 }
 
+function getAllRelationshipsByArtist(req, res) {
+    _getAllRelationShipsBy(req, res, relationships.mapArtistToSubject);
+}
+
+function getAllRelationshipsByWork(req, res) {
+    _getAllRelationShipsBy(req, res, relationships.mapWorkToSubject);
+}
+
+function _getAllRelationShipsBy(req, res, mappingFunction) {
+    var params = req.swagger.params;
+    var id = params.id.value;
+    var offset = params.offset ? params.offset.value : undefined;
+    var limit = params.limit ? params.limit.value : undefined;
+    var subject = mappingFunction(id);
+    var relation = params.relation ? params.relation.value : undefined;
+    var object = params.object ? params.object.value : undefined;
```

Comment:
You can directly use `params.object.value` which evaluates to `undefined` if not given as parameter.